EXILED/Exiled.Events/Patches/Fixes/VoiceChatTpsFix.cs

```diff
@@ -0,0 +1,43 @@
+// -----------------------------------------------------------------------
+// <copyright file="VoiceChatTpsFix.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.Events.Patches.Fixes
+{
+    using System.Collections.Generic;
+    using System.Reflection.Emit;
+
+    using API.Features.Pools;
+
+    using HarmonyLib;
+
+    using VoiceChat.Networking;
+
+    using static HarmonyLib.AccessTools;
+
+    /// <summary>
+    /// Fixes <see cref="VoiceTransceiver.ServerReceiveMessage(Mirror.NetworkConnection, VoiceChat.Networking.VoiceMessage)"/> method.
+    /// </summary>
+    [HarmonyPatch(typeof(VoiceTransceiver), nameof(VoiceTransceiver.ServerReceiveMessage))]
+    internal static class VoiceChatTpsFix
+    {
+        private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions, ILGenerator generator)
+        {
+            List<CodeInstruction> newInstructions = ListPool<CodeInstruction>.Pool.Get(instructions);
+            int index = newInstructions.FindIndex(instruction => instruction.Calls(Method(typeof(UnityEngine.Mathf), nameof(UnityEngine.Mathf.Abs), new System.Type[] { typeof(float) }))) + 11;
+            newInstructions.InsertRange(index, new CodeInstruction[]
+            {
+                new CodeInstruction(OpCodes.Pop),
+                new CodeInstruction(OpCodes.Ldc_I4_S, 480),
+            });
```

Comment:
Firstly, i think indexing can be improved, 11 for such method is a strange offset, i think you can find a better instructions to reduce an offset.
Secondly, i dont understand your logic, you inject your instuctions in compile-generated `foreach` code. I think you wanted to do a different thing, or am I missing something?
Can you explain on human-lang what the patch is all about?
<img width="527" alt="image" src="https://github.com/user-attachments/assets/5a573470-4231-4655-90b5-2f8ab29632c5">

Reply:
uh okay, youre pops `_playerPackets.Length` and loads `480` instead of it. for foreach of `480` elements instead of all array? it actually looks very strange.
It certainly won't break the anticheat?

Reply:
atleast better indexing would be:
```cs
int offset = -2;
int index = newInstructions.FindIndex(i => i.opcode == OpCodes.Ldloc_2) + offset;
```

Reply:
<img width="269" alt="image" src="https://github.com/user-attachments/assets/4dbbb157-8eab-43f2-be78-4aff813fbb00">

Reply:
No, it wont break ac. Ok il change indexing. What am i doing: VoiceMessage is 480p long, but array of playerpackets  is 24000 length, meaning when decoding message into it, there is 23520 not used for iterations.

Reply:
I Edited Patch A Little Bit, now it works by changine array size when it is created.

Reply:
oh, now i understand